vktrace/vktrace_layer/vktrace_lib_pageguard.cpp

```diff
@@ -494,15 +494,14 @@ VkResult vkFlushMappedMemoryRangesWithoutAPICall(VkDevice device, uint32_t memor
     if (!g_trimEnabled) {
         // trim not enabled, send packet as usual
         FINISH_TRACE_PACKET();
-    } else if (g_trimIsPreTrim) {
-        vktrace_finalize_trace_packet(pHeader);
-    } else if (g_trimIsInTrim) {
-        // Currently tracing the frame, so need to track references & store packet to write post-tracing.
+    } else {
```

Comment:
The changes in file vktrace_lib_pageguard.cpp seem like they are unrelated to the gist of this pull request. If it is unrelated, could you make this change in a separate pull request? Or maybe I'm wrong?

Reply:
You are correct.  It is not fixing the issue of "unlock uninitialized mutex lock" but a fix to the risk of "lock on a locked mutex lock".  I forgot to update the description of the pull request when pushing the later two commits.  Shall I update the description of this pull request or create a new one?

I think it is better to have both fixes being merged together since both of them are needed to fix the mutex lock problem.

Reply:
Please create a new pull request.

Reply:
Sure, https://github.com/LunarG/VulkanTools/pull/519 has been created.